scripts/serverless.js

```diff
@@ -125,6 +125,34 @@ const processSpanPromise = (async () => {
             // Hack to not duplicate the warning with similar deprecation
             logDeprecation.triggeredDeprecations.add('VARIABLES_ERROR_ON_UNRESOLVED');
           }
+        } else if (!isHelpRequest) {
+          if (variablesMeta.has('provider')) {
+            throw new ServerlessError(
+              `Cannot resolve ${path.basename(
+                configurationPath
+              )}: "provider" section is not accessible (configured behind variables which cannot be resolved at this stage)`
```

Comment:
Do we document somewhere which variable sources are supported at this stage? Users might be wondering which types of variables cannot be resolved at this stage

Reply:
This is very picky, and in the end we may end with many stages, but I hope naturally in most cases users will not approach those errors (it may happen only when doing some weird things, as putting `provider` behind e.g. some AWS dependent var).

So goal is, to make knowledge of this stages not be considered as important to work with the Framework

Still I believe it'll be nice to have some variable resolution anatomy document for those who want to inspect deep details.

Reply:
Makes sense - thanks for clarification, as you mentioned, it might be useful to have that documentation for those interested but I don't think now is good time to add that :+1: